spec/models/application_draft_spec.rb

```diff
@@ -37,17 +57,48 @@
 
       end
 
-      context 'with two students' do
+      Student::REQUIRED_DRAFT_FIELDS.each do |attribute|
+        let(:ace_student)     { double.as_null_object }
+        let(:value)           { SecureRandom.hex(12) }
+
         before do
-          allow(subject).to receive(:students).and_return([failing_student])
+          allow(ace_student).to receive(attribute).and_return(value)
         end
 
-        context 'one being invalid' do
-          skip
+        context 'with two students' do
+          context 'one being invalid' do
+            before do
+              allow(subject).to receive(:students).and_return([failing_student, ace_student])
+            end
+
+            it "is satisfied when the corresponding student #{attribute} is set" do
+              expect { subject.valid? :apply }.not_to \
+                change { subject.errors["student1_#{attribute}"] }
+            end
+
+            it "requires student0_#{attribute} for 'apply' context" do
+              expect { subject.valid? :apply }.to \
+                change { subject.errors["student0_#{attribute}"] }.to include "can't be blank"
+            end
+
+          end
         end
 
         context 'both being valid' do
-          skip
+          let(:ace_student_2)   { double.as_null_object }
+          let(:value_2)         { SecureRandom.hex(12)  }
+
+          before do
+            allow(subject).to receive(:students).and_return([ace_student, ace_student_2])
```

Comment:
You should be able to re-use the `ace_student` from up above as we don't have a mechanism in place that would check for the students not being the same reference. If it works w/o, lines 88-90 can be removed as well.